### PR TITLE
Add warning for missing attention mask when pad tokens are detected to various models

### PR DESCRIPTION
**Pull request by hackyon**
# What does this PR do?

This PR builds upon #24510 by adding the warning to many more models (and also a template files). The original issue with more context can be found at #16136.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@gante @ydshieh

### CODE REVIEW

**Comment by hackyon** [comment]
Let me know if this makes sense, thanks!

**Comment by ydshieh** [comment]
Thank you for the PR @hackyon .

We don't maintain the files under the directory `examples/research_projects`. Could you revert the changes in it?
 Otherwise LGTM.

**Comment by ydshieh** [comment]
There seems to some strange torch test issues though. I will take a look.

**Comment by hackyon** [comment]
I think the failing test is due to the newly added method, I'll take a closer look into it tomorrow.

**Comment by gante** [comment]
It looks good to me, will approve after the tests are fixed :)

**Comment by hackyon** [comment]
The test was failing since the __contains__ operation on the input_ids were not supported by FX/JIT tracing (context on tracing here: https://pytorch.org/docs/stable/fx.html)

To fix the test, I opted to simply skip the check if tracing is enabled. This should be fine since the check doesn't actually alter any functionality and doesn't really need to be traced.

Please take another look, thanks!

**Comment by ydshieh** [comment]
Still good for me, thanks!

**Review by ydshieh — APPROVED** [review]
Thanks again!

**Review by sgugger — APPROVED** [review]
Thanks a lot for adding it everywhere!

**Comment by ydshieh** [comment]
Sorry @gante I forgot your comment 

> will approve after the tests are fixed :)

**Comment by gante** [comment]
nw, it looks good to me 👍
